clients/go/admin/auth_interceptor.go

```diff
@@ -50,6 +52,21 @@ func shouldAttemptToAuthenticate(errorCode codes.Code) bool {
 	return errorCode == codes.Unauthenticated
 }
 
+// Set up http client used in oauth2
+func setHTTPClientContext(ctx context.Context, cfg *Config) context.Context {
+	httpClient := &http.Client{}
```

Comment:
and there's no issue with the overhead of creating these clients right?  if not specified the default client in the http lib gets used it looks like.

Reply:
if not specified it will just be a `&http.Client{}`